src/cmap/wire_protocol/on_demand/document.ts

```diff
@@ -329,14 +339,24 @@ export class OnDemandDocument {
    * Deserialize this object, DOES NOT cache result so avoid multiple invocations
    * @param options - BSON deserialization options
    */
-  public toObject(options?: BSONSerializeOptions): Record<string, any> {
-    return BSON.deserialize(this.bson, {
+  public toObject(options?: OnDemandDocumentDeserializeOptions): Record<string, any> {
+    return deserialize(this.bson, {
       ...options,
       index: this.offset,
       allowObjectSmallerThanBufferSize: true
     });
   }
 
+  private parseBsonSerializationOptions(options?: { enableUtf8Validation?: boolean }): {
```

Comment:
is this still necessary? it looks like we are just calling the bson parseUtf8ValidationOptions now

Reply:
This is "unused" according to vscode graying it out, should be good to remove